spec/requests/providers/application_merits_task/opponent_organisations_controller_spec.rb

```diff
@@ -0,0 +1,161 @@
+            expect(response.body).to include("Organisation type")
+          end
+        end
+
+        context "when unauthenticated" do
+          before { get_new_opponent }
+
+          it_behaves_like "a provider not authenticated"
+        end
+      end
+
+      describe "show: GET /providers/applications/:legal_aid_application_id/opponent_organisations/:opponent_id" do
+        subject(:get_existing_opponent) { get providers_legal_aid_application_opponent_organisation_path(legal_aid_application, opponent) }
+
+        let(:opponent) { create(:opponent, :for_organisation, legal_aid_application:) }
+
+        context "when authenticated" do
+          before do
+            login_provider
+            get_existing_opponent
+          end
+
+          it "returns success" do
+            expect(response).to have_http_status(:ok)
+          end
+
+          it "displays opponent's details" do
+            expect(response.body).to include(html_compare(opponent.name))
+            expect(response.body).to include(html_compare(opponent.description))
+          end
+        end
+
+        context "when unauthenticated" do
+          before { get_existing_opponent }
+
+          it_behaves_like "a provider not authenticated"
+        end
+      end
+
+      describe "update: PATCH /providers/applications/:legal_aid_application_id/opponent_organisations/:opponent_id" do
+        subject(:patch_organisation) do
+          patch(
+            providers_legal_aid_application_opponent_organisation_path(legal_aid_application, opponent),
+            params: params.merge(button_clicked),
+          )
+        end
+
+        let!(:opponent) { create(:opponent, :for_organisation, legal_aid_application:, organisation_name: "Mid Bedfordshire Council", organisation_ccms_code: "GOVT", organisation_description: "Government Department") }
+
+        let(:params) do
+          {
+            application_merits_task_opponent: {
+              name: "Central Bedfordshire Council",
+              organisation_type_ccms_code: "LA",
+            },
+          }
+        end
+
+        let(:draft_button) { { draft_button: "Save as draft" } }
+        let(:button_clicked) { {} }
+
+        before do
+          allow(LegalFramework::MeritsTasksService).to receive(:call).with(legal_aid_application).and_return(smtl)
+          login_provider
+        end
+
+        it "amends the opponent opposable organisation with the values entered" do
+          expect { patch_organisation }.not_to change(::ApplicationMeritsTask::Organisation, :count)
+          opponent.opposable.reload
+          expect(opponent.opposable.name).to eql "Central Bedfordshire Council"
+          expect(opponent.opposable.ccms_code).to eql "LA"
+        end
+
+        it "sets the task to complete" do
+          patch_organisation
+          expect(legal_aid_application.legal_framework_merits_task_list.serialized_data).to match(/name: :opponent_name\n\s+dependencies: \*\d+\n\s+state: :complete/)
```

Comment:
Ditto on state
```ruby
expect(legal_aid_application.legal_framework_merits_task_list).to have_completed_task(:application, :opponent_name)
```
it may need a reload on the `legal_aid_application` object

Reply:
👍 have made suggested change